lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,311 @@
+      doSearch(newVal);
+    }
+  }
+
+  function doSearch(val: string) {
+    focusTextInput();
+    onSearch(val.trim());
+  }
+
+  function focusTextInput() {
+    inputRef.current?.querySelector('input')?.focus();
+  }
+
+  // helper for focusing different buttons in the dropdown menu
+  const adjustBtnFocus = (adjust: (i: number) => number) => () => {
+        const newFocusableBtnIndex = adjust(focusableBtnIndex ?? 0),
+            elToFocus = dropdownRef.current?.children[newFocusableBtnIndex] as HTMLElement | null;
+
+        if (elToFocus) {
+          setFocusableBtnIndex(newFocusableBtnIndex);
+          elToFocus.scrollIntoView({ block: 'nearest' });
+          setInlineStyle(false);
+        }
+      },
+      focusNext = adjustBtnFocus(inc),
+      focusPrev = adjustBtnFocus(dec),
+      focusFirst = adjustBtnFocus(always(0)),
+      focusLast = adjustBtnFocus(always(matches.length - 1));
+
+  function handleOnClick({displayName}: DataItem<T, string>) {
+    onChange(displayName);
+    onSearch(displayName.trim());
+    focusTextInput();
+    setFocusableBtnIndex(null);
+    setInlineStyle(false);
+  }
+
+  function handleKeyUp(evt: KeyboardEvent<HTMLElement>) {
+    if (autoComplete) {
+      const isPrintableChar = evt.key.length === 1 && evt.key.match(/[!-~]/);
+      setInlineStyle(!!isPrintableChar);
+    }
+  }
+
+  function handleKeyDown(evt: KeyboardEvent<HTMLElement>) {
+    const inputEle = evt.currentTarget as HTMLInputElement,
+        elToFocusText = matches.length && focusableBtnIndex !== null ? matches[focusableBtnIndex].displayName : null,
+        endIndex = autoComplete && elToFocusText ? elToFocusText.length : value.length;
+
+    switch (evt.key) {
+      case 'Enter':
+        if (focusableBtnIndex !== null) {
+          handleOnClick(matches[focusableBtnIndex]);
+          inputEle.setSelectionRange(endIndex, endIndex);
+        }
+        evt.preventDefault();
+        break;
+      case 'Home':
+        inputEle.setSelectionRange(0, 0);
+        evt.preventDefault();
+        break;
+      case 'End':
+        inputEle.setSelectionRange(endIndex, endIndex);
+        evt.preventDefault();
+        break;
+      case 'ArrowDown':
+        if (focusableBtnIndex === null || focusableBtnIndex === matches.length - 1) {
+          focusFirst();
+        }
+        else {
+          focusNext();
+        }
+        evt.preventDefault();
+        break;
+      case 'ArrowUp':
+        if (focusableBtnIndex) {
+          focusPrev();
+        }
+        else {
+          focusLast();
+        }
+        evt.preventDefault();
+        break;
+      case 'Escape':
+        handleOnChange('');
+        setFocusableBtnIndex(null);
+        break;
+      case 'Backspace':
+        if (inlineStyle && value !== elToFocusText) {
+          evt.preventDefault();
+          setFocusableBtnIndex(null);
+        }
+        setInlineStyle(false);
+        break;
+      case ' ':
+        setInlineStyle(false);
```

Comment:
In the example, if I type "Item ", I lose autocomplete even though that is a valid prefix of some options.  It doesn't seem like we'd want that, so why is this line here?

Reply:
> this makes me think maybe the value passed down to onSearch should not be trimmed in order to check if it is a valid prefix?

You know what I think you're right about that.  Do you think it should be totally untrimmed or have the front trimmed but not the back?

Reply:
I think it should be totally untrimmed so the length will be correct when `setSelectionRange`.

Reply:
Updated here 307a8d35c12fcd54b1dbf88c491aaa1f8bf64089.